### PR TITLE
[kitchen] Add RHEL 8 to platforms tested

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Adds RHEL 8 to platforms tested in kitchen tests.

### Motivation

Kitchen tests on all supported platforms.

### Additional Notes

Depends on a [chef-datadog PR](https://github.com/DataDog/chef-datadog/pull/641) to add support for RHEL 8, and on a builders update to update `chef` and `kitchen` in the kitchen runners.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ variables:
   S3_DSD6_URI: s3://dsd6-staging/linux
   RELEASE_VERSION: nightly
   DATADOG_AGENT_BUILDIMAGES: v1631023-f2ae59b
-  DATADOG_AGENT_BUILDERS: v1613952-f072482
+  DATADOG_AGENT_BUILDERS: v1694687-834dd8c
```
Comment (KSerrania) [inline .gitlab-ci.yml]: Will be changed once the builders update is merged.

File: test/kitchen/Berksfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 source 'https://supermarket.chef.io'
 
-cookbook 'datadog', '~> 3.2.0'
+cookbook 'datadog', git: 'https://github.com/DataDog/chef-datadog.git', branch: 'kserrania/rhel-8-support'
```
Comment (KSerrania) [inline test/kitchen/Berksfile]: Will be reverted once the chef-datadog PR is merged.

**Review by remeh — DISMISSED** [review]
Feel free to merge when you'll have updated the tag and the cookbook dependency. 🚀

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1065,6 +1065,7 @@ kitchen_centos-a6:
     - rsync -azr --delete ./ $SRC_PATH
     - export TEST_PLATFORMS="centos-69,OpenLogic:CentOS:6.9:6.9.20180530"
     - export TEST_PLATFORMS="$TEST_PLATFORMS|centos-76,OpenLogic:CentOS:7.6:7.6.20190402"
+    - export TEST_PLATFORMS="$TEST_PLATFORMS|rhel-80,RedHat:RHEL:8:8.0.2019062020"
```
Comment (remeh) [inline .gitlab-ci.yml]: We must add this to the `kitchen_centos-a7` job now.

**Review by remeh — DISMISSED** [review]
A last small addition is needed following the appearance of the a7 jobs. Feel free to merge after the change / having validated that it works in a pipeline. 👍

**Review by remeh — APPROVED** [review]

**Pull request merged**